decentraland/unity-renderer
# chore: rpc scene controller service

### WHY
We are migrating kernel <-> renderer messages to the new RPC paradigm

### WHAT

- Implemented Scene Controller RPC service
    - `LoadScene()`: SDK6 and SDK7 scenes now request scenes loading through this RPC
    - `UnloadScene()`: now SDK6 and SDK7 scenes are unloaded using this RPD
    - `SendCrdt()`: SDK7 scenes send and receive CRDT messages using this RPC
    - `GetCurrentState()`
    - Filtering to only register service on scene ports
    - Scene auto-unload if its port is closed

[Kernel counterpart PR](https://github.com/decentraland/kernel/pull/755) (won't be merged until this unity-renderer PR has reached production).

### QA TESTING
Please test this PR with a regular Release Candidate test plan to make sure these communication changes don't break anything that may have missed by the implementator.

* **Web Client build to test**: https://play.decentraland.zone/?renderer-branch=chore/rpc-scene-controller-service
* **Desktop client build to test**: https://github.com/decentraland/explorer-desktop/pull/442
